dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/FilePermissionsServiceIT.java

```diff
@@ -0,0 +1,233 @@
+import edu.harvard.iq.dataverse.authorization.groups.GroupServiceBean;
+import edu.harvard.iq.dataverse.datafile.FilePermissionsService;
+import edu.harvard.iq.dataverse.engine.command.exception.PermissionException;
+import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
+import edu.harvard.iq.dataverse.persistence.group.Group;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.GuestUser;
+import edu.harvard.iq.dataverse.persistence.user.NotificationType;
+import edu.harvard.iq.dataverse.persistence.user.RoleAssignment;
+import edu.harvard.iq.dataverse.persistence.user.UserNotification;
+import edu.harvard.iq.dataverse.persistence.user.UserNotificationDao;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import static edu.harvard.iq.dataverse.mail.FakeSmtpServerUtil.waitForEmailSentTo;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author madryk
+ */
+@RunWith(Arquillian.class)
+public class FilePermissionsServiceIT extends WebappArquillianDeployment {
+
+    @Inject
+    private FilePermissionsService filePermissionsService;
+
+    @Inject
+    private DataFileServiceBean dataFileService;
+    @Inject
+    private AuthenticationServiceBean authenticationService;
+    @Inject
+    private GroupServiceBean groupService;
+    @Inject
+    private DataverseSession dataverseSession;
+    @Inject
+    private RoleAssigneeServiceBean roleAssigneeService;
+    @Inject
+    private UserNotificationDao userNotificationDao;
+    
+    
+    @Before
+    public void initBefore() {
+        dataverseSession.setUser(authenticationService.getAuthenticatedUser("superuser"));
+    }
+    
+    
+    // -------------------- TESTS --------------------
+    
+    @Test
+    public void assignFileDownloadRole() {
+        // given
+        DataFile datafile1 = dataFileService.find(53L);
+        DataFile datafile2 = dataFileService.find(55L);
+        AuthenticatedUser user = authenticationService.getAuthenticatedUser("superuser");
+        Group group = groupService.getGroup("explicit/1-rootgroup");
+        
+        datafile1.getFileAccessRequesters().add(user);
+        datafile1 = dataFileService.save(datafile1);
```

Comment:
I'm assuming here that `dataFileService.save()` is an dao method (no business logic). Probably it would me more clear if we would have already clear separation between dao and service layer.
Using EntityManager directly will be kind of troublesome here (we would have to add transaction management)